### DISCUSSION TITLE
I built a passwordless auth example

### DISCUSSION THREAD

**Discussion by byoungdale** [mojolicious/mojo.js#97] [Show and tell]
I wanted to share [mojojs-passwordless-example](https://github.com/byoungdale/mojojs-passwordless-example). I used it to learn mojojs, and I figured I would post it here in case anyone else finds it useful as an example.

Also, any feedback is welcome. I wasn't trying to build anything "production" ready, but it could still be useful to make any changes that folks recommend.

Thanks for the fun framework!

**Comment by kraih** [comment]
Thanks for sharing the example app, much appreciated. As for feedback, what jumps out at me is `bg-worker.js` and `migrate.js`. The former i would merge into the main web app, make the email task a plugin, and start the worker with `node lib/index.js minion-worker -j 12`. Still have to port the documentation for this organisational pattern from Perl to JavaScript. And the latter could just be an [application command](https://mojojs.org/docs/Cookbook.md#adding-commands-to-your-application) (`src/cli/migrate.ts` called as `node lib/index.js migrate ...`).

**Reply by byoungdale** [reply]
thank you! I knew there had to be a better way with those two.

**Comment by byoungdale** [comment]
@kraih I updated the example app so it is now using [plugins](https://mojojs.org/docs/Cookbook.md#adding-plugins-to-your-application) and [applications commands](https://mojojs.org/docs/Cookbook.md#adding-commands-to-your-application). Hopefully other folks find it useful when getting started.

**Reply by kraih** [reply]
Is there a reason you're separating Minion from the web app? Normally i would expect mojo.js apps to use the approach described in the [Minion.js docs](https://github.com/mojolicious/minion.js#mojojs) with `minionPlugin`.

**Reply by byoungdale** [reply]
Ah, no particular reason. I must not have read the docs closely enough. I didn’t realize the minion plug-in integrates it into the web app.

**Reply by kraih** [reply]
I see. In case you missed it, there's also an [example app](https://github.com/mojolicious/minion.js/tree/main/examples/linkcheck) included with Minion.js that might be useful for some inspiration.

**Reply by kraih** [reply]
Stuff like using a [separate PostgreSQL schema per test](https://github.com/mojolicious/minion.js/blob/155a5cb3155bccfa27ac93a0796ef419bbea9c3d/examples/linkcheck/test/linkcheck.js) comes to mind, which allows for faster parallel testing by isolating each test environment.

**Reply by byoungdale** [reply]
thank you! i had missed that. I only found the [example](https://github.com/mojolicious/pg.js/tree/main/examples/blog) in the pg.js repo, which has also been helpful.
